### DISCUSSION TITLE
How to print a `div` with style

### DISCUSSION THREAD

**Discussion by SongOfYouth** [unoplatform/uno#18735] [Q&A] [answered]
I've tried a js method:
```js
function PrintDiv(ID) {
    var needPrint = document.getElementById(ID);
    var head = document.head.innerHTML;
    var win = window.open("");

    win.document.write('<html><head>' + head + '</head><body>'
        + needPrint.outerHTML + '</body>'
        + "</html>");
    win.document.head.removeAttribute('theme-color');
    win.document.close();
    //Chrome
    if (navigator.userAgent.indexOf("Chrome") != -1) {
        win.onload = function () {
            win.document.execCommand('print');
            win.close();
        }
    }
    //Firefox
    else {
        win.print();
        win.close();
    }
}
```
but it dosen't display correctly.
the page content:
![image](https://github.com/user-attachments/assets/a4456ff8-9971-4748-bf0a-121adb9bab19)
the print page content:
![image](https://github.com/user-attachments/assets/70e37ac1-54e6-40b8-a222-1c823d16a0e0)
It's gray and not display the content fully.

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the question. At this time, printing is not something we've invested in, and it's unclear what happens when doing so. Manipulating the DOM directly is likely to cause additional issues of rendering.
